uarray/backend.py

```diff
@@ -9,11 +9,62 @@
 ArgumentReplacerType = Callable[[Tuple, Dict, Tuple], Tuple[Tuple, Dict]]
 
 
-class BackendNotImplementedError(TypeError):
-    pass
+class BackendNotImplementedError(NotImplementedError):
+    """
+    An exception that is thrown when no compatible backend is found for a method.
+    """
 
 
 class MultiMethod:
+    """
+    This class is meant to provide a dispatchable method with or without a registered implementation.
+
+    Provides a multi-method instance. The method has no implementation by default, but one can be provided
+    if needed. :obj:`Backend` s can provide implementations for the method, and users can choose backends.
+
+    Parameters
+    ----------
+    argument_extractor : ArgumentExtractorType
+        This is a callable that extracts the arguments using which dispatch can be performed from this
```

Comment:
I had to read this more than once, so propose a small tweak: "... extracts the arguments that will be used to dispatch over". 

It would also help to state somewhere what happens if this returns more than one parameter. Do they all need to match? E.g. for the `potato` example, if `a` and `b` are both array kinds, does it ever make sense to use `b` (I think not)? And if `b` is a dtype for example, it would only dispatch to registered methods where both array and dtype match, and raise an error otherwise?